homeassistant/components/omnilogic/omnilogic_common.py

```diff
@@ -0,0 +1,127 @@
+"""Common classes and elements for Omnilogic Integration."""
+
+from datetime import timedelta
+import logging
+
+from omnilogic import OmniLogicException
+
+from homeassistant.components.sensor import ENTITY_ID_FORMAT
+from homeassistant.const import ATTR_NAME
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+    UpdateFailed,
+)
+
+from .const import ATTR_IDENTIFIERS, ATTR_MANUFACTURER, ATTR_MODEL, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class OmniLogicUpdateCoordinator(DataUpdateCoordinator):
+    """Class to manage fetching update data from single endpoint."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        api: str,
+        name: str,
+        polling_interval: int,
+    ):
+        """Initialize the global Omnilogic data updater."""
+        self.api = api
+
+        super().__init__(
+            hass=hass,
+            logger=_LOGGER,
+            name=name,
+            update_interval=timedelta(seconds=polling_interval),
+        )
+
+    async def _async_update_data(self):
+        """Fetch data from OmniLogic."""
+        try:
+            _LOGGER.debug("Updating the coordinator data.")
+            data = await self.api.get_telemetry_data()
+            return data
+
+        except OmniLogicException as error:
+            raise UpdateFailed("Error updating from OmniLogic: %s" % error) from error
+
+
+class OmniLogicEntity(CoordinatorEntity):
+    """Defines the base OmniLogic entity."""
+
+    def __init__(
+        self,
+        coordinator: DataUpdateCoordinator,
+        kind: str,
+        name: str,
+        backyard: dict,
+        bow: dict,
+        icon: str,
+        entitydata: dict,
+    ):
+        """Initialize the OmniLogic Entity."""
+        super().__init__(coordinator)
+
+        if bow != {}:
+            entityname = (
+                backyard["BackyardName"].replace(" ", "_")
+                + "_"
+                + bow["Name"].replace(" ", "_")
+                + "_"
+                + kind
+            )
+        else:
+            entityname = backyard["BackyardName"].replace(" ", "_") + "_" + kind
+
+        self._kind = kind
+        self._name = None
```

Comment:
Name is set within the updates since there are several different types of sensor which would have different naming and additional platforms will also have unique naming. At the master class level we don't want to specify the name and rather leave that to the platforms themselves.